README.md

```diff
@@ -1 +1,241 @@
 # avr-libstdcpp: libstdc++ port for avr-gcc
+
+`avr-libstdcpp` is a partial, non-fully-tested
+implementation of the C++ standard library and its STL.
+It is intended to be used with `avr-gcc`.
+
+Many features of modern C++11,14,17 and 20 are supported.
+
+`avr-gcc` does not have a complete C++ standard library nor does it include an STL implementation.
+The `avr-libstdcpp` port (even though not-fully-tested
+and only partially full/complete) will, nonetheless,
+be useful for those interested in making more comprehensive utilization
+of C++ --- including its standard library --- with a modern `avr-gcc` compiler.
+
+## Historical origins
+
+The `avr-libstdcpp` library traces its own origins to existing
+GNU/GCC C++ standard library implementaion(s) --- themselves targeting
+embedded systems. This work is essentially an even-more
+_embedded_-_friendly_ adaptation of the aforementioned work.
+
+The `avr-libstdcpp` port began in 2018 with an initial import
+of a GNU/GCC-based C++ standard library from GCC 8.
+A second import of a GNU/GCC-based C++ standard library from GCC 10
+in 2020 modernized the port to include many modern C++20 features.
+
+## Using the library
+
+Add the `avr-libstdcpp/include` path to the standard `-I`
+include path(s) of the compiler on the command line.
+Upon doing this, include standard library headers in the usual way.
+
+For example:
+
+Simply add the `-I` include path to your particular location
+of `avr-libstdcpp/include` on the command line...
+
+```sh
+avr-g++ -O2 -x c++ -I/my_path/avr-libstdcpp/include -mmcu=atmega328p test.cpp -o test.elf
+```
+
+... and seamlessly use standard library headers in your code.
+
+```C
```

Comment:
```suggestion
```cpp
```

Reply:
Oh that's great. That MD detail had eluded me for toooo long. Thx